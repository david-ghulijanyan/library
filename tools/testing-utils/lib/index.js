import * as jestDom from "@testing-library/jest-dom";
import * as reactHooks from "@testing-library/react-hooks";
import * as userEvent from "@testing-library/user-event";
import * as api from "./api";
// import * as data from "./data";
import * as mocks from "./mocks";
import * as spies from "./spies";
import * as utils from "./utils";

const { react } = utils;

const def = {
	api,
	// data,
	mocks,
	spies,
	...utils,
	jestDom,
	reactHooks,
	userEvent,
};

export default def;

export { api, mocks, spies, react, jestDom, reactHooks, userEvent };
