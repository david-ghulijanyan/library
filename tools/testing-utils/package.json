{
	"name": "@library/testing-utils",
	"version": "0.1.0",
	"description": "Testing Utils",
	"main": "dist/cjs/index.js",
	"module": "dist/esm/index.js",
	"private": false,
	"author": {
		"email": "david.hayki.ghulijanyan@gmail.com",
		"name": "David Ghulijanyan"
	},
	"license": "UNLICENSED",
	"engines": {
		"node": ">=16.13.0",
		"npm": ">=8.1.0"
	},
	"files": [
		"dist"
	],
	"publishConfig": {
		"registry": "http://localhost:4873"
	},
	"scripts": {
		"build": "rimraf dist && npm run build:esm && npm run build:cjs",
		"postbuild2": "npm run clean:dist",
		"build:esm": "cross-env NODE_ENV=production BABEL_ENV=esm babel ./lib --copy-files --out-dir 'dist/esm' --source-maps --ignore 'node_modules,**/node_modules'",
		"build:cjs": "cross-env NODE_ENV=production BABEL_ENV=cjs babel ./lib --copy-files --out-dir 'dist/cjs' --source-maps --ignore 'node_modules,**/node_modules'",
		"clean:dist": "rimraf $(find ./dist -regextype egrep -regex \".*/__snapshots__|.*/__tests__|.*.spec.js$|.*.test.js$\")",
		"start": "node scripts/start.js",
		"test": "node scripts/test.js"
	},
	"eslintConfig": {
		"extends": [
			"@library/eslint-config"
		]
	},
	"prettier": "@library/prettier-config",
	"browserslist": {
		"production": [
			">0.2%",
			"not dead",
			"not op_mini all"
		],
		"development": [
			"last 1 chrome version",
			"last 1 firefox version",
			"last 1 safari version"
		]
	},
	"peerDependencies": {
		"axios": "^0.23.0",
		"react": "^17.0.2",
		"react-dom": "^17.0.2"
	},
	"dependencies": {
		"@testing-library/jest-dom": "^5.14.1",
		"@testing-library/react": "^12.1.2",
		"@testing-library/react-hooks": "^7.0.2",
		"@testing-library/user-event": "^13.5.0",
		"babel-jest": "^27.3.1",
		"faker": "^5.5.3",
		"jest": "^27.3.1",
		"jest-circus": "^27.3.1",
		"jest-cli": "^27.3.1",
		"jest-prop-type-error": "^1.1.0",
		"jest-resolve": "^27.3.1",
		"jest-watch-typeahead": "^1.0.0",
		"msw": "^0.35.0"
	},
	"devDependencies": {
		"@babel/cli": "^7.15.7",
		"@babel/core": "^7.15.8",
		"@library/babel-preset": "file:../../tools/babel-preset",
		"@library/eslint-config": "file:../../tools/eslint-config",
		"@library/prettier-config": "file:../../tools/prettier-config",
		"cross-env": "^7.0.3",
		"rimraf": "^3.0.2"
	}
}
