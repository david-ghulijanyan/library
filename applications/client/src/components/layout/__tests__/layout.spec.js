import { react as utils } from "@library/testing-utils";
import Layout from "./wrapper";

const { render } = utils;

describe("Layout component", () => {
	test("Should render properly and match snapshot", () => {
		// Arrange
		// Act
		const { container } = render(Layout, { children: 1 });

		// Assert
		expect(container).toMatchSnapshot();
	});
	test("Should not render properly if no children provided", () => {
		// Arrange
		const error = jest.spyOn(console, "error").mockReturnValue(true);
		// Actw
		const { container } = render(Layout);

		// Assert
		expect(error).toBeCalledTimes(1);
		expect(container).toMatchSnapshot();
	});
});
