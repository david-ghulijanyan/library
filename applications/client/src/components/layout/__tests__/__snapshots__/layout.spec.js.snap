// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Should not render properly if no children provided 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1futezt-MuiTypography-root"
        >
          Library
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        />
        <a
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root css-1y942vo-MuiButtonBase-root-MuiButton-root"
          href="/signin"
          tabindex="0"
        >
          Sing In
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root css-1y942vo-MuiButtonBase-root-MuiButton-root"
          href="/signup"
          tabindex="0"
        >
          Sign Up
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg main-wrapper css-1oqqzyl-MuiContainer-root"
  />
</div>
`;

exports[`Layout component Should render properly and match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1futezt-MuiTypography-root"
        >
          Library
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        />
        <a
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root css-1y942vo-MuiButtonBase-root-MuiButton-root"
          href="/signin"
          tabindex="0"
        >
          Sing In
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root css-1y942vo-MuiButtonBase-root-MuiButton-root"
          href="/signup"
          tabindex="0"
        >
          Sign Up
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg main-wrapper css-1oqqzyl-MuiContainer-root"
  >
    1
  </main>
</div>
`;
